### PR TITLE
open tracking

### PR DESCRIPTION
**Pull request by rileylnapier**
## Description

send open tracking for messages rendered

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> `Fix [#1]()`

### CODE REVIEW

File: packages/react-inbox/src/components/Message/index.tsx [hunk 1]
```diff
@@ -78,6 +79,14 @@ const Message: React.FunctionComponent<
     }
   };
 
+  useEffect(() => {
```
Comment (rileylnapier) [inline packages/react-inbox/src/components/Message/index.tsx:82]: might be better to use a "in view" hook vs "rendered"

File: packages/react-inbox/src/hooks/use-on-screen.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { useEffect } from "react";
+
+export const useOnScreen = (ref, cb) => {
+  useEffect(() => {
```
Comment (suhasdeshpande) [inline packages/react-inbox/src/hooks/use-on-screen.ts:4]: would naming function would aid developer experience?

File: packages/react-toast/src/components/Toast/index.tsx [hunk 3]
```diff
@@ -49,9 +49,21 @@ export const Toast: React.FunctionComponent<
...
           : message;
 
-      toast(<Body {...message} icon={message?.icon ?? config?.defaultIcon} />, {
-        role: config?.role ?? "status",
-      });
+      toast(
+        <Body
+          {...message}
+          icon={
+            typeof message?.icon === "string"
```
Comment (suhasdeshpande) [inline packages/react-toast/src/components/Toast/index.tsx:56]: why can't we callback to either `message?.icon || config.defaultIcon`? This is a bit hard to read and I am not sure if this is a by-product of some product specification.

**Review by suhasdeshpande — APPROVED** [review]
overall fine. I am a bit foggy on some logic and it is verbose.

File: packages/react-toast/src/components/Toast/index.tsx (hunk 3 above)
Comment (rileylnapier) [reply packages/react-toast/src/components/Toast/index.tsx:56]: the type made me do it like this, you are right

**Pull request merged**